src/main/java/frc/robot/subsystems/FalconDriveSubsystem.java

```diff
@@ -185,4 +159,40 @@ public void stop() {
   private double metersToEncTicks(final double m) {
     return m * ENC_TICKS_PER_METER;
   }
+
+  /**
+   * 
+   * @param et The number of encoder ticks to be converted to meters.
+   * @return The number of meters that correspond to @param m , assuming 6-inch
+   *         wheels.
+   */
+  private double encTicksToMeters(final double et) {
+    return et / ENC_TICKS_PER_METER;
+  }
+
+  // Updates the PID constants stored in each Talon FX from MoPrefs
+  private void updatePIDConstants() {
+    double kP = MoPrefs.getDriveKP();
+    double kI = MoPrefs.getDriveKI();
+    double kD = MoPrefs.getDriveKD();
+    int kIZ = (int) MoPrefs.getDriveKIZ();
+    double kFF = MoPrefs.getDriveKFF();
+    leftFront.config_kP(0, kP);
```

Comment:
This would be easy if the Preferences class allowed us access to the underlying `NetworkTableEntry`s.  Unfortunately, that is not the case.
Also, references is a final class, so we can't subclass it and give ourselves access.

I can see two options:
1. Re-implement the Preferences class, and add a method to get access to the `NetworkTableEntry`. The Preferences class is pretty simple, so this wouldn't be too difficult. It does seem a bit hacky though.
2. Create some sort of "watcher" class that watches a value, looks for changes, and triggers a lambda when the value changes.  This would be pretty trivial to implement, however it would require polling the Preferences NetworkTable on every loop. This is probably not a big deal, though, and I suspect the NetworkTables are designed to support it.

I'm not sure which option is better.

Reply:
I thought MoPrefs already subclassed Preferences.  Apparently, it does not, so it must wrap it instead (composition vs inheritance).

I also thought there was a way to subscribe to NT entries, but if there's not, then having a mechanism in place to detect and notify on changes would certainly be helpful.

Reply:
What if instead of polling Preferences and writing the values to the Talon FXs every scheduler cycle, we only did that once per second?  That would be trivially easy to implement and would basically accomplish the same goal.

Reply:
It might help to understand the root problem.
The SparkMax docs say specifically that certain values like the PID settings shouldn't be written frequently.  I suspect the TalonFX has a similar warning.
The reason is because those values are held in a form of non-volatile memory (flash or eeprom) which has a limited number of write cycles in its lifetime.  You want to make sure you only write that when you're changing it.  Writing the same value repeatedly only wears out the storage and shortens the life of the device.

Theoretically, this type of check should be happening in the vendor code between you and the device, but you can never know for sure and if the docs say don't do it, then it's a safer bet to assume they aren't checking for you.

Reply:
There is a way to subscribe to Networktable entries. However, the WPILib Preferences class doesn't let us access those entries. Just the values.